docs/decision/03-data-storage.md

```diff
@@ -0,0 +1,105 @@
+# Data Storage
+
+Status: Draft
+
+Date: 2020-05-22
+
+Author: Filipe Pinheiro <filipe@yld.io>
+
+Reviewed-by: Tim Seckinger <tim.seckinger@yld.io>
+
+## Context
+
+The ASAP Hub has to store data about users and usage of the application. To make that decision, we need to take into consideration the data model and data access patterns the application needs.
+
+## Data Model
+
+### Entity Relational Diagram
+
+```
+erDiagram
+  User ||..|| Invite : has
+  User ||..o{ Auth : has
+```
+
+https://mermaid-js.github.io/mermaid-live-editor/
+[![](https://mermaid.ink/img/eyJjb2RlIjoiZXJEaWFncmFtXG4gIFVzZXIgfHwuLnx8IEludml0ZSA6IGhhc1xuICBVc2VyIHx8Li5veyBBdXRoIDogaGFzIiwibWVybWFpZCI6eyJ0aGVtZSI6ImRlZmF1bHQifSwidXBkYXRlRWRpdG9yIjpmYWxzZX0)](https://mermaid-js.github.io/mermaid-live-editor/#/edit/eyJjb2RlIjoiZXJEaWFncmFtXG4gIFVzZXIgfHwuLnx8IEludml0ZSA6IGhhc1xuICBVc2VyIHx8Li5veyBBdXRoIDogaGFzIiwibWVybWFpZCI6eyJ0aGVtZSI6ImRlZmF1bHQifSwidXBkYXRlRWRpdG9yIjpmYWxzZX0)
+
+### Data access patterns
+
+**Invite (one time token)**
```

Comment:
Not sure these will actually be stored. They can probably just be an HMAC over the email or something. Either way, I think we can omit this part - even if we store them, they will be stored like any other '**Auth**' method

Reply:
Being an HMAC over the email makes them guessable, the usage as another `auth` method sounds good.

Reply:
How does it make them guessable? We're obviously not going to give anyone the HMAC secret :sweat_smile:

Reply:
Diagonal reading. We need to be able to revoke the token and possibly generate new ones for the same email.

Reply:
Okay. There are multiple ways of implementing the revocation, but this could indeed be a reason to store the tokens. I'll just add a 'potential' to the headline I guess so we're safe.